rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m

```diff
@@ -0,0 +1,171 @@
+        self.openInBrowserButton
+    ];
+}
+
+- (UIBarButtonItem *)webBarButtonItemWithImage:(UIImage *)image action:(SEL)selector {
+    return [[[UIBarButtonItem alloc] initWithImage:image
+                                             style:UIBarButtonItemStylePlain
+                                            target:self.webView
+                                            action:selector] autorelease];
+}
+
+// MARK: - Lazy
+
+- (WKWebView *)webView {
+    if(!_webView) {
+        _webView = [WKWebView new];
+        _webView.navigationDelegate = self;
+        _webView.translatesAutoresizingMaskIntoConstraints = NO;
+    }
+    return _webView;
+}
+
+- (UIBarButtonItem *)goBackButton {
+    if(!_goBackButton) {
+        _goBackButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"arrow-narrow-left"]
+                                                  action:@selector(goBack)] retain];
+    }
+    return _goBackButton;
+}
+
+- (UIBarButtonItem *)goForwardButton {
+    if(!_goForwardButton) {
+        _goForwardButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"arrow-narrow-right"]
+                                                     action:@selector(goForward)] retain];
+    }
+    return _goForwardButton;
+}
+
+- (UIBarButtonItem *)reloadWebPageButton {
+    if(!_reloadWebPageButton) {
+        _reloadWebPageButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"refresh"]
+                                                         action:@selector(reload)] retain];
+    }
+    return _reloadWebPageButton;
+}
+
+- (UIBarButtonItem *)closeWebPageButton {
+    if(!_closeWebPageButton) {
+        _closeWebPageButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"xmark"]
+                                                        action:@selector(closeWebPage)] retain];
+    }
+    return _closeWebPageButton;
+}
+
+- (UIBarButtonItem *)openInBrowserButton {
+    if(!_openInBrowserButton) {
+        _openInBrowserButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"safari"]
+                                                                style:UIBarButtonItemStylePlain
+                                                               target:self
+                                                               action:@selector(openInBrowser)];
+    }
+    return _openInBrowserButton;
+}
+
+// MARK: -
+
+- (void)closeWebPage {
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (void)openInBrowser {
+    [UIApplication.sharedApplication openURL:self.webView.URL
+                                     options:@{}
+                           completionHandler:^(BOOL success) {
+        [self closeWebPage];
+    }];
+}
+
+// MARK: - WKNavigationDelegate
+
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction
+decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
+    switch (navigationAction.navigationType) {
```

Comment:
What is purpose of this method?

Reply:
without implementing this method of delegate webview doesn't know how to deal with hyperlinks 
[read the docs pls](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455641-webview?language=objc)

Reply:
This method isn't mandatory to implement. Web view opens links without implementing it. You did not introduce any decide policy inside - in any case you call `decisionHandler(WKNavigationActionPolicyAllow);`

Reply:
idk, without it links don't work

Reply:
Well I think that actually links are works, but there is no indication of loading process of webpage, and it seems as not working(no progress, navigation arrows didn'n change state, some kind of activity indicator etc). I pulled changes from branch and now can't check this:)

Reply:
Resolved. It fixes opening any links which have `target="_blank"` attribute in their HTML <`a href`> tag.

Reply:
wow 😅✋